src/renderer/components/+namespaces/namespace-select-filter.tsx

```diff
@@ -91,27 +95,33 @@ export class NamespaceSelectFilter extends React.Component<SelectProps> {
     disposeOnUnmount(this, [
       reaction(() => this.isMenuOpen, newVal => {
         if (newVal) { // rising edge of selection
-          this.selected.replace(namespaceStore.selectedNames);
+          if (namespaceStore.areAllSelectedImplicitly) {
+            this.selected.replace([""]);
+          } else {
+            this.selected.replace(namespaceStore.selectedNames);
+          }
           this.didToggle = false;
         }
       }),
     ]);
   }
 
   formatOptionLabel({ value: namespace, label }: SelectOption) {
+    let isSelected = false;
+
     if (namespace) {
-      const isSelected = namespaceStore.hasContext(namespace);
-
-      return (
-        <div className="flex gaps align-center">
-          <Icon small material="layers" />
-          <span>{namespace}</span>
-          {isSelected && <Icon small material="check" className="box right" />}
-        </div>
-      );
+      isSelected = !namespaceStore.areAllSelectedImplicitly && namespaceStore.hasContext(namespace);
+    } else {
+      isSelected = namespaceStore.areAllSelectedImplicitly;
```

Comment:
maybe this can be simplified? I just think it is odd to assign to `false` and then override it completely.

```ts
const isSelected = namespace
  ? !namespaceStore.areAllSelectedImplicitly && namespaceStore.hasContext(namespace)
  : namespaceStore.areAllSelectedImplicitly
```

Reply:
Yes, will cleanup this.